### ISSUE TITLE
[Request] Consider a "legacy" build for compatibility.

### ISSUE THREAD

**Issue by johnth3doe** [polymorphicshade/NewPipe#56]
Greetings to one and all.

It has come to my attention that:
1) "OkHttp 2.12.x is being discontinued on December 31, 2021", and also that 
2) "media2 only supports API 21 and newer, so that would require dropping KitKat support".
[Kindly see "Reference A"]
Bearing this is mind, there is also a "legacy" version of newpipe that _is_ compatible (though I am unaware the finer details) for the select few (including myself) that are still on KitKat devices; would like to request that this be kindly taken into account when new versions/builds are made. 
I would personally suggest either making relevant changes to the main builds that would keep older devices compatible.

Thank you for your time and consideration. Stay safe and be well.

Reference:
A) https://github.com/TeamNewPipe/NewPipe/pull/5457

Edited: Wrong version number was quoted in the initial draft.

**Comment by johnth3doe** [comment]
@CalAlaera: Would you be so kind as to explain as to why you disagree/down-voted?

Note: I did say "I am unaware of the finer details", so if you only disagree with my (un-techn[olog]ically aware) opinion or that I overstepped in something I stated, then to each their own, but I would be glad if you would educate me in the understanding that I lack. 

However, if this is not the case, the next logical interpretation would be that you disagree in a **request** that allows a small group of affected users to still use NP (+Sponsorblock) to maintain their privacy and (possibly) security... In which case it suggests that your reaction is unamicable.

That being said, I apologize in advance (even though it is in the bottom of the comment) if I seem hostile in this response, I would like it to be known that my statement is with no ill-will.

Thank you for your time and consideration.

**Comment by CalAlaera** [comment]
@johnth3doe I'm not sure why you felt the need to call me out by name, and I would have preferred to keep my opinions to myself. But, since you asked, I reacted the way I did primarily for two reasons:
1. The market share for vintage operating systems such as the one you specified appears to be significantly less than 1% of active device users [(source)](https://www.statista.com/statistics/921152/mobile-android-version-share-worldwide/). With this in mind, and given that the developer has limited time they are prepared to spend on this project, I felt their time could be better spent on issues that affect all users rather than a tiny subset of users.
2. The vintage operating system in question has not received security patches for several years [(source)](https://source.android.com/security/bulletin/2017-10-01). There are several known security breaches which will never be patched, and any new ones won't be patched either, including most notably an unpatched security flaw that renders the OS vulnerable to man-in-the-middle attacks [(source)](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0818). Anyone using it as their mobile OS therefore has **much** bigger problems than not being able to watch YouTube videos.

**Comment by johnth3doe** [comment]
> @johnth3doe I'm not sure why you felt the need to call me out by name, and I would have preferred to keep my reasons to myself. But, since you asked, I reacted the way I did primarily for two reasons:
> 
>     1. The market share for vintage operating systems such as the one you specified appears to be significantly less than 1% of active device users [(source)](https://www.statista.com/statistics/921152/mobile-android-version-share-worldwide/). With this in mind, and given that the developer has limited time they are prepared to spend on this project, I felt their time could be better spent on issues that affect all users rather than a tiny subset of users.
> 
>     2. The vintage operating system in question has not received security patches for several years [(source)](https://source.android.com/security/bulletin/2017-10-01). There are several known security breaches which will never be patched, and any new ones won't be patched either, including most notably an unpatched security flaw that renders the OS vulnerable to man-in-the-middle attacks [(source)](http://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2016-0818). Anyone using it as their mobile OS therefore has **much** bigger problems than not being able to watch YouTube videos.

@CalAlaera: Before I respond, I would like to thank you for your patience and apologize for any inconvenience.

Firstly, I called you out by name as you were - and (at the time of this comment) still are - specifically the one that reacted to my initial statement, and as for you wanting to withhold your opinion, I requested an explanation and appreciate that you went out of your way to provide one.

To address the points mentioned:
1) I am (as mentioned) one of the affected (in this case "1%") and when reading the page from the  link that I referenced in the initial statement I came across "https://github.com/TeamNewPipe/NewPipe/pull/5457#issuecomment-765412828" and had assumed that if-and/or-when the official newpipe made a 'commit' or 'gradle' change (I don't know anything about these things), it would be possible to mirror the change or something to that effect. I admittedly may not have made this clear initially and take full responsibility for that.
2) I am glad that you brought these issues and vulnerabilities to my attention, I am however unfortunately (as stated above) one of the aforementioned with (as you put it) "much bigger problems" and cannot (for personal reasons) do much about it. therefore desired to at least keep trying to have a good thing while it lasts and was even possible. Admittedly this is not the burden of the developer nor the majority; and I didn't intend to cause any unnecessarily complex matters. Moreover, to be clear, I never associated NewPipe with being the penultimate in cybersecurity nor even contemplated the developer to involve their valuable effort to that effect.

I would like to reiterate that mine was (and still is) a polite request; nothing more, nothing less. I don't expect for anyone to go above and beyond for anything that they do not desire or are unable to; but as long as it is not harmful, asking is the least that one should be able to do.  At least without being downvoted or dismissed without any apparent or known reason.

In conclusion, while I welcome your personal opinion on how you feel that the developer should be spending their limited time, and also praise you for educating me as I had asked; I shall kindly and hopefully leave this issue open to the consideration  of the developer, and let the developer themselves ultimately decide if they are able or willing to entertain a respectful and humble request.

To one and all; stay safe, be well.
Thank you for your time and consideration.
